File: app/src/main/java/edu/application/ui/fragments/settings/ColorSchemeCustomFragment.java
package edu.application.ui.fragments.settings;

import static android.content.Context.MODE_PRIVATE;

import android.content.SharedPreferences;
import android.graphics.drawable.ColorDrawable;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;

import edu.application.databinding.FragmentColorsSchemeCustomBinding;
import edu.application.ui.adapters.TextFormatter;

public class ColorSchemeCustomFragment extends Fragment {

    private FragmentColorsSchemeCustomBinding binding;
    private SharedPreferences sharedPreferences;
    private SharedPreferences.Editor editor;

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater,
                             @Nullable ViewGroup container,
                             @Nullable Bundle savedInstanceState) {

        binding = FragmentColorsSchemeCustomBinding.inflate(inflater, container, false);

        sharedPreferences = requireActivity().getPreferences(MODE_PRIVATE);
        editor = sharedPreferences.edit();

        binding.picker.addSVBar(binding.svBar);
        binding.picker.addOpacityBar(binding.opacityBar);

        TextFormatter.setColor(binding.cvText);
        TextFormatter.setBorder(binding.sampleTextCv);
        TextFormatter.setTextSize(binding.cvText);

        return binding.getRoot();
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);

        binding.backBtn.setOnClickListener(v -> requireActivity().onBackPressed());

        binding.setBackgroundColorBtn.setOnClickListener(v -> {
            binding.picker.setOldCenterColor(binding.picker.getColor());
            binding.cvText.setBackgroundColor(binding.picker.getOldCenterColor());
        });
        binding.setTextColorBtn.setOnClickListener(v -> {
            binding.picker.setOldCenterColor(binding.picker.getColor());
            binding.cvText.setTextColor(binding.picker.getColor());
            binding.sampleTextCv.setStrokeColor(binding.picker.getColor());
        });

        binding.saveBtn.setOnClickListener(v -> {
            editor.putInt("lesson_background_color",
                    ((ColorDrawable) binding.cvText.getBackground()).getColor());
            editor.putInt("lesson_text_color", binding.cvText.getCurrentTextColor());
            editor.apply();
        });
    }
}

File: app/src/main/java/edu/application/ui/fragments/settings/ColorSchemeCustomFragment.kt
package edu.application.ui.fragments.settings

import android.content.SharedPreferences
import android.graphics.drawable.ColorDrawable
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import edu.application.databinding.FragmentColorsSchemeCustomBinding
import edu.application.ui.adapters.TextFormatter.setBorder
import edu.application.ui.adapters.TextFormatter.setColor
import edu.application.ui.adapters.TextFormatter.setTextSize
import edu.application.ui.adapters.TextFormatter.sharedPreferences

class ColorSchemeCustomFragment : Fragment() {

    private lateinit var binding: FragmentColorsSchemeCustomBinding
    private lateinit var editor: SharedPreferences.Editor

    override fun onCreateView(inflater: LayoutInflater,
                              container: ViewGroup?,
                              savedInstanceState: Bundle?): View {
        binding = FragmentColorsSchemeCustomBinding.inflate(inflater, container, false)
        editor = sharedPreferences.edit()

        binding.picker.addSVBar(binding.svBar)
        binding.picker.addOpacityBar(binding.opacityBar)

        setColor(binding.cvText)
        setBorder(binding.sampleTextCv)
        setTextSize(binding.cvText)

        return binding.root
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)

        binding.backBtn.setOnClickListener { requireActivity().onBackPressed() }

        binding.setBackgroundColorBtn.setOnClickListener {
            binding.picker.oldCenterColor = binding.picker.color
            binding.cvText.setBackgroundColor(binding.picker.oldCenterColor)
        }
        binding.setTextColorBtn.setOnClickListener {
            binding.picker.oldCenterColor = binding.picker.color
            binding.cvText.setTextColor(binding.picker.color)
            binding.sampleTextCv.strokeColor = binding.picker.color
        }

        binding.saveBtn.setOnClickListener {
            editor.putInt("lesson_background_color",
                (binding.cvText.background as ColorDrawable).color)
            editor.putInt("lesson_text_color", binding.cvText.currentTextColor)
            editor.apply()
        }
    }
}